README.md

```diff
@@ -1,32 +1,254 @@
+
+```
+cd ${TOOLKIT_DIR}
+java -cp build/libs/gapic-generator-*-fatjar.jar com.google.api.codegen.configgen.ConfigGeneratorTool \
+  --descriptor_set=${YOUR_DESCRIPTOR_FILE} --service_yaml=${YOUR_SERVICE_YAML} \
+  -o=${YOUR_CLIENT_CONFIG}
+```
+
+You can safely ignore the warning about control-presence.
+
+### Manually tweak the generated client config
+
+The generated client config contains FIXME comments with instructions on how to choose values
+in the client config. The client config should work as-is though; tweaks are only necessary to improve
+the quality of the generated output.
+
+### Create a package metadata config file
+
+You need to locate/decide on the following before you create the package metadata config file:
+
+1. The output file to contain the package metadata for your client generation. Hereafter, this will be
+   referenced as `${PKG_META_CONFIG}`.
+2. A short name, which typically does not include the major version. For example, Cloud Language v1's short name
+   is `language`. Hereafter, this will be referenced as `${SHORT_NAME}`.
+3. The major version. The first major version will typically be `v1`. Hereafter, this will be referenced as
+   `${MAJOR_VERSION}`.
+4. The package name, which typically includes the fully-qualified product name and major version.
+   For example, Cloud Language v1's package name is `google-cloud-language-v1`. Hereafter, this will be
+   referenced as `${PACKAGE_NAME}`.
+5. The proto path, which is the proto's package with dots converted to slashes. For example, Cloud Language v1's
+   proto path is `google/cloud/language/v1`. Hereafter, this will be referenced as `${PROTO_PATH}`.
+
+First, copy the contents of api_defaults.yaml and dependencies.yaml into a single file, hereafter
+referenced as `${PKG_META_CONFIG}`.
+
+```
+cp ${GOOGLEAPIS_DIR}/gapic/packaging/api_defaults.yaml ${PKG_META_CONFIG}
+cat < ${GOOGLEAPIS_DIR}/gapic/packaging/dependencies.yaml >> ${PKG_META_CONFIG}
+```
+
+Add the following lines to the end:
+
+```
+artifact_type: GAPIC
+proto_deps:
+- google-common-protos
+short_name: ${SHORT_NAME}
+major_version: ${MAJOR_VERSION}
+package_name:
+  default: ${PACKAGE_NAME}
+proto_path: ${PROTO_PATH}
+```
+
+### Run code generation
+
+You need to locate/decide on the following before you call code generation:
+
+1. The language-specific generator config for your desired output language. Given a language where
+   `${language}` is in `["csharp", "go", "java", "nodejs", "php", "py", "ruby"]`, the file is located at
+   `src/main/resources/com/google/api/codegen/${language}/${language}_gapic.yaml` (exception: the Python file is named
+   `python_gapic.yaml`). Hereafter, this will be referenced as `${LANGUAGE_CONFIG}`.
+2. The output directory for your generated client classes. Hereafter, this will be
+   referenced as `${GENERATED_CLIENT_DIR}`.
+
+```
+java -cp build/libs/gapic-generator-*-fatjar.jar com.google.api.codegen.CodeGeneratorTool \
```

Comment:
I think we should considering adding `.sh/.bat` runner scripts around toolkit for this purpose. It seems like a industry standard of doing it for java apps (for example both gradlew and IntelliJ IDEA do it this way). Btw, this may allow us to avoid adding fatJar task in gradle (correct classpath can be handled in the sh script itself).

Reply:
It's considerably easier to distribute a fat jar - it's actually somewhat difficult to grab all of the jars out of the gradle cache. 

I'm fine with adding `.sh`/`.bat` files later, but I don't want to tackle it at the moment.